### ISSUE TITLE
Time Hierarchy Theorem

### ISSUE THREAD

**Issue by monadplus** [mofeing/computational-complexity#1]

**Comment by mofeing** [comment]
It would be nice a chapter about both time and space hierarchy theorems. I know we also have a lecture on the non-deterministic model that overlapped with this, but I prefer to have a seperate lecture.

What do you think?

**Comment by jproyo** [comment]
I would love to have it too, because there are some things that are tricky there.

**Comment by monadplus** [comment]
I will work on it next week because I need to finish first the 1st delivery for Advance Data Structures.

Notify me If I forget about this!

**Comment by monadplus** [comment]
BTW, some of these topics are really interesting.

Here is a short blog-post about why non-turing-completeness is a positive feature (in some) programming languages (http://www.haskellforall.com/2020/01/why-dhall-advertises-absence-of-turing.html). It's a 5' reading ;)

**Comment by mofeing** [comment]
@monadplus We can separate them and speak about space hierarchy in another chapter. Did we speak about the space hierarchy theorem in class? I can't remember.

Great, no hurry. But before starting anything, let's define the different parts first. #2

**Comment by jproyo** [comment]
> @monadplus We can separate them and speak about space hierarchy in another chapter. Did we speak about the space hierarchy theorem in class? I can't remember.
Yes, we have. Obviously notes maybe differ from Book but it has been discussed on class.
 
> Great, no hurry. But before starting anything, let's define the different parts first. #2

I am on the same position as @monadplus. Perhaps during next week i can dedicate some time to enrich this doc.
